tests/database/test_journal_db.py

```diff
@@ -163,6 +201,78 @@ def test_committing_middle_changeset_merges_in_subsequent_changesets(journal_db)
     assert journal_db.journal.has_changeset(changeset_c) is False
 
 
+def test_squash_does_not_persist_0_checkpoints(journal_db, memory_db):
+    journal_db.set(b'1', b'test-a')
+
+    # should have no effect
+    journal_db.squash()
+
+    assert b'1' not in memory_db
+    assert b'1' in journal_db
+
+    journal_db.persist()
+
+    assert b'1' in memory_db
+
+
+def test_squash_does_not_persist_1_checkpoint(journal_db, memory_db):
+    journal_db.set(b'1', b'test-a')
+
+    checkpoint = journal_db.record()
+
+    journal_db.set(b'2', b'test-b')
```

Comment:
I really liked you're other pr that used semantically meaningful keys.

```python
journal_db.set((b'in-base-checkpoint', ...)

checkpoint = journal_db.record()

journal_db.set(b'in-first-checkpoint', ...)

...
```

It made the later assertions a lot easier to read without having to re-evaluate the whole test body.